src/Jobs/UpdatePriceJob.php

```diff
@@ -46,9 +47,19 @@ public function handle(ChecksMagentoExistence $checksMagentoExistence): void
 
         $data = $model->getData();
 
-        $this->handleBasePrices($data);
-        $this->handleTierPrices($data);
-        $this->handleSpecialPrices($data);
+        $batch = Bus::batch(array_filter([
+            $this->handleBasePrices($data),
+            $this->handleTierPrices($data),
+            $this->handleSpecialPrices($data),
+        ]))
+            ->dispatch();
+
+        if (! $batch->hasFailures()) {
```

Comment:
Does this work when the batch is run via the queue? How can it know of the batch has failures before it has finished running?

Shouldn't you use the `then` callback so that it resets the state when the batch is finished?